dbms/src/AggregateFunctions/AggregateFunctionSum.cpp

```diff
@@ -29,59 +32,69 @@ using AggregateFunctionSumSimple = AggregateFunctionSum<T, typename NearestField
 template <typename T>
 using AggregateFunctionCountSecondStage = AggregateFunctionSum<T, typename NearestFieldType<T>::Type, AggregateFunctionSumData<typename NearestFieldType<T>::Type>, NameCountSecondStage>;
 
-template <typename T, typename TResult>
-using AggregateFunctionSumDecimal = AggregateFunctionSum<T, TResult, AggregateFunctionSumData<TResult>>;
+template <typename T, typename TResult, typename SumName>
+using AggregateFunctionSumDecimal = AggregateFunctionSum<T, TResult, AggregateFunctionSumData<TResult>, SumName>;
 
 template <typename T>
-using AggregateFunctionSumWithOverflow = AggregateFunctionSum<T, T, AggregateFunctionSumData<T>>;
+using AggregateFunctionSumWithOverflow = AggregateFunctionSum<T, T, AggregateFunctionSumData<T>, NameSumWithOverFlow>;
 
 template <typename T>
-using AggregateFunctionSumKahan = AggregateFunctionSum<T, Float64, AggregateFunctionSumKahanData<Float64>>;
+using AggregateFunctionFinalSumStage = AggregateFunctionSum<T, T, AggregateFunctionSumData<T>, NameFinalSumStage>;
```

Comment:
As the comments below, what we want to add is not `FinalSumStage`, instead, in agg function level, we just wants to add a agg sum function that the result type is equal to its input type, it is exactly what `SumWithOverFlow` do, so why not just reuse `SumWithOverFlow`? And in order to make it less confusing, we can just add a name alias like `SumOnPartialReults` for it.

Reply:
> As the comments below, what we want to add is not `FinalSumStage`, instead, in agg function level, we just wants to add a agg sum function that the result type is equal to its input type, it is exactly what `SumWithOverFlow` do, so why not just reuse `SumWithOverFlow`? And in order to make it less confusing, we can just add a name alias like `SumOnPartialReults` for it.

okk